### ISSUE TITLE
using word static in class doesn't report error

### ISSUE THREAD

**Issue by tombohub** [microsoft/TypeScript#54291]
### 🕗 Version & Regression Information

version 5.0.4

### ⏯ Playground Link

this code runs ok:

```javascript
class DateOnly {
  static parse(s: string): DateOnly {
    return s;
  }
}
```
[playgroung](https://www.typescriptlang.org/play?ts=5.0.4#code/PTAEFkE9QYwewCYFNQHM5IM6gBZIE5ICwAUDADYCGm2AIpQC5IDyAdudAN6miiYOMAljFAAHSvkxIAFJgBcfBvkGtUASgX0mbDqG4levQgwCu+VnwDcPUAF9StoA)

this code now shows error:

```typescript
class DateOnly {
  static parse(s: string): DateOnly {
    return s;
  }
  kok() {
    return "koko";
  }
}
```

[playground2](https://www.typescriptlang.org/play?ts=5.0.4#code/PTAEFkE9QYwewCYFNQHM5IM6gBZIE5ICwAUDADYCGm2AIpQC5IDyAdudAN6miiYOMAljFAAHSvkxIAFJgBcfBvkGtUASgX0mbDqG4levQgwCu+VnwDcPUAF8bAazgPpavTaNJT50ACInTr7WBnaktkA)

adding static word again doesn't show the error:

```typescript
class DateOnly {
  static parse(s: string): DateOnly {
    return s;
  }
  static kok() {
    return "koko";
  }
}
```

[playground3](https://www.typescriptlang.org/play?ts=5.0.4&ssl=9&ssc=2&pln=2&pc=1#code/PTAEFkE9QYwewCYFNQHM5IM6gBZIE5ICwAUDADYCGm2AIpQC5IDyAdudAN6miiYOMAljFAAHSvkxIAFJgBcfBvkGtUASgX0mbDqG4levQgwCu+VnwDcPUAF8b-ISIDWcZ9LV6bRpKfOgAIldXAOsDO1JbIA)

### 🙁 Actual behavior

obvious type error is not reported

### 🙂 Expected behavior

to error on type error

**Comment by MartinJohns** [comment]
This is working as intended. The error is not due to the use of `static`, but because your class is structurally the same as `string`, so you it's valid to return `s`. Once your class has actual instance members (like `kok` in your second playground) the class is not structurally compatible anymore.

See also: [Why do these empty classes behave strangely?
](https://github.com/microsoft/TypeScript/wiki/FAQ#why-do-these-empty-classes-behave-strangely)
